src/runtime_intrinsics.c

```diff
@@ -342,16 +344,22 @@ jl_value_t *jl_iintrinsic_1(jl_value_t *ty, jl_value_t *a, const char *name,
         memset((char*)pa2 + isize, getsign(pa, isize), osize2 - isize);
         pa = pa2;
     }
-    jl_value_t *newv = lambda1(ty, pa, osize, osize2, list);
-    if (ty == (jl_value_t*)jl_bool_type)
-        return *(uint8_t*)jl_data_ptr(newv) & 1 ? jl_true : jl_false;
-    return newv;
+    if (jl_is_bitstype(aty) && jl_is_bitstype(ty)) {
+        jl_value_t *newv = lambda1(ty, pa, osize, osize2, list);
+        if (ty == (jl_value_t*)jl_bool_type)
+            return *(uint8_t*)jl_data_ptr(newv) & 1 ? jl_true : jl_false;
+        return newv;
+    } else if (jl_is_vec_type(aty) && jl_is_vec_type(ty)) {
```

Comment:
After discussion with @vtjnash yesterday evening we came to the conclusion that it might be worth refactoring runtime_intrinsics, because it is very hard to extend right now.